### DISCUSSION TITLE
specFiltering examples help

### DISCUSSION THREAD

**Discussion by jurjandrei** [webdriverio/webdriverio#10814] [Q&A]
Hello,

I am looking on skipping tests before the webdriver session is started using Mocha and found about https://webdriver.io/blog/page/6/

I was curios if there are some examples on how     featureFlags: {
        specFiltering: true
    }, is taken in consideration and how should annotate a test in order to be skipped?
Also is there a possibility to apply conditions for when to skip the tests? 
For example a test should be skipped only when two conditions are met.

Thanks

**Comment by christian-bromann** [comment]
Hey @jurjandrei ,
this is an old blog post and the feature is now automatically built-in. If the testrunner detects that it doesn't have to run any tests, it won't start a session.

**Reply by jurjandrei** [reply]
Hey @christian-bromann 
Unfortunately we are still on v4 and wanted to see if it works also on this version?
How will the test runner detect if and what tests needs to run? Should there be some tags in place for each test? Thanks

**Reply by christian-bromann** [reply]
I strongly recommend updating to the latest version. We have deprecated v4 many years ago and it actually surprises me that you can still run tests with it.

**Reply by jurjandrei** [reply]
Yeah, we are working on updating as quickly as possibile. But even after upgrading, it is still unclear on how the test runner will skip the tests, so if there is some documentation which eluded me, please do share it.

**Reply by christian-bromann** [reply]
I am not sure but the testrunner in v8 detects whether the test framework , e.g. Mocha, would run any tests and if not it doesn't create any session and just closes the worker.

**Comment by jurjandrei** [comment]
@mgrybyk maybe you know the answer to this? Saw that you wrote this article: https://webdriver.io/blog/2019/11/01/spec-filtering/

I am curios on how the filtering is done. 

Much appreciated
